packages/logging/src/analytics-helpers.ts

```diff
@@ -275,7 +273,10 @@ export class ThrottledAnalytics implements MongoshAnalytics {
     if (this.currentUserId) {
       throw new Error('Identify can only be called once per user session');
     }
-    this.currentUserId = message.userId ?? message.anonymousId;
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    this.currentUserId = message.userId ?? message.anonymousId!;
```

Comment:
Unfortunately because of the more derivative type I think TypeScript can no longer correctly resolve this scenario